sycl/test/basic_tests/vectors/assign.cpp

```diff
@@ -0,0 +1,84 @@
+// TODO: Remove `__SYCL_USE_PREVIEW_VEC_IMPL` once it's auto-set.
+// RUN: %clangxx -fsycl -fsyntax-only %s -fpreview-breaking-changes -D__SYCL_USE_PREVIEW_VEC_IMPL=1
+// RUN: %clangxx -fsycl -fsyntax-only %s
+
+#include <sycl/sycl.hpp>
+
+using sycl::half;
+using sycl::vec;
+using sw_half_1 = decltype(std::declval<vec<half, 4>>().swizzle<1>());
+using sw_half_2 = decltype(std::declval<vec<half, 4>>().swizzle<1, 2>());
+
+using sw_float_1 = decltype(std::declval<vec<float, 4>>().swizzle<1>());
+using sw_float_2 = decltype(std::declval<vec<float, 4>>().swizzle<1, 2>());
+
+using sw_double_1 = decltype(std::declval<vec<double, 4>>().swizzle<1>());
+using sw_double_2 = decltype(std::declval<vec<double, 4>>().swizzle<1, 2>());
+
+#if __INTEL_PREVIEW_BREAKING_CHANGES
+#define NOT_IN_PREVIEW !
```

Comment:
NIT; how about we rename this macro to something like `NEGATE_IF_NOT_IN_PREVIEW` ?

Reply:
I don't see how it would help, honestly. This isn't "! in preview", but rather simply English, like "hey this holds, except not in preview".

Reply:
> This isn't "! in preview", but rather simply English, like "hey this holds, except not in preview".

Initially, I had the same confusion. That's why I'd have renamed this macro to `NEGATE_IF_NOT_IN_PREVIEW` or something. Anyway, I don't have strong opinion on this.